config/routes.rb

```diff
@@ -20,9 +20,13 @@
     resource :vote, controller: 'comments/votes', only: [:create, :destroy]
   end
 
-  get ':username' => 'users/lists#index', as: :user_profile
+  get ':reference_id/:id/delete' => 'links#destroy', as: :destroy_link
+  get ':username/:list_id/:id/update' => 'references#edit_abstract', as: :edit_abstract
```

Comment:
> For the edit_abstract, I kept it as a 'get' because it's just calling for the abstract form instead of updating

That's correct

> editing the whole reference instead of just the abstract

Quick question, we are editing the paper and not the abstract, correct?

> I read through http://guides.rubyonrails.org/routing.html but it didn't seem to give any advice about actions for editing a specific attribute.

That's because you can edit a single attribute through an action designed to edit the whole entity.